src/Graphics/Image/Processing/Filter.hs

```diff
@@ -140,3 +141,78 @@ prewittOperator !img = sqrt (prewittX ^ (2 :: Int) + prewittY ^ (2 :: Int))
+--
+laplacianFilter :: (Array arr cs e, Array arr X e) =>
+                   Direction -> Border (Pixel cs e) -> Filter arr cs e
+laplacianFilter dir !border =
+  Filter (correlate border kernel)
+  where
+    !kernel =
+      case dir of
+        Complete   -> fromLists $ [ [ -1, -1, -1 ]     -- Unlike the Sobel edge detector, the Laplacian edge detector uses only one kernel. 
+                                  , [  -1, 8, -1 ]     -- It calculates second order derivatives in a single pass. 
+                                  , [  -1, -1, -1 ]]   -- This is the approximated kernel used for it. (Includes diagonals)
+{-# INLINE laplacianFilter #-}
+
+-- | 'Laplacian of Gaussian' (LOG) filter is a two step process of smoothing
+-- an image before applying some derivative filter on it. This comes in
+-- need for reducing the noise sensitivity while working with noisy
+-- datasets or in case of approximating second derivative measurements.
+-- 
+-- The LoG operator takes the second derivative of the image. Where the image  
+-- is basically uniform, the LoG will give zero. Wherever a change occurs, the LoG will
+-- give a positive response on the darker side and a negative response on the lighter side.
+-- 
+-- <<images/yield.jpg>>   <<images/yield_log.png>> 
+--
+logFilter :: (Array arr cs e, Array arr X e) =>
+             Direction -> Border (Pixel cs e) -> Filter arr cs e
+logFilter dir !border =
+  Filter (correlate border kernel)
+  where
+    !kernel =
+      case dir of
+        Vertical   -> fromLists $ [ [ 0, 1, 1, 2, 2, 2, 1, 1, 0 ]              
+                                  , [  1,  2,  4, 5, 5, 5, 4, 2, 1 ]
+                                  , [  1,  4,  5, 3, 0, 3, 5, 4, 1 ] 
+                                  , [  2,  5,  3, -12, -24, -12, 3, 5, 2 ] 
+                                  , [  2,  5,  0, -24, -40, -24, 0, 5, 2  ] 
+                                  , [  2,  5,  3, -12, -24, -12, 3, 5, 2  ] 
+                                  , [  1,  4,  5, 3, 0, 3, 5, 4, 1  ] 
+                                  , [  1,  2,  4, 5, 5, 5, 4, 2, 1  ] 
+                                  , [  0,  1,  1, 2, 2, 2, 1, 1, 0  ]] 
+
+        Horizontal -> fromLists $ [ [ 0, 1, 1, 2, 2, 2, 1, 1, 0 ]
+                                  , [  1,  2,  4, 5, 5, 5, 4, 2, 1 ]
+                                  , [  1,  4,  5, 3, 0, 3, 5, 4, 1 ] 
+                                  , [  2,  5,  3, -12, -24, -12, 3, 5, 2 ] 
+                                  , [  2,  5,  0, -24, -40, -24, 0, 5, 2  ] 
+                                  , [  2,  5,  3, -12, -24, -12, 3, 5, 2  ] 
+                                  , [  1,  4,  5, 3, 0, 3, 5, 4, 1  ] 
+                                  , [  1,  2,  4, 5, 5, 5, 4, 2, 1  ] 
+                                  , [  0,  1,  1, 2, 2, 2, 1, 1, 0  ]] 
+
+{-# INLINE logFilter #-}
+
+logOperator :: (Array arr cs e, Array arr X e, Floating e) => Image arr cs e -> Image arr cs e
```

Comment:
This function essentially computes `sqrt(2)*abs(LoG)`, but does so in computationally expensive way :)
You also should have gotten a compile error since `INLINE sobelOperator` pragma doesn't match the function name.
In any case, I don't think `logOperator` is a useful function in Image processing.

Reply:
Yes sir. Very sorry. I did my testing using another file. Will correct it.